File: tools/token_bench/programs/05_http_api/http_api.py
from flask import Flask, request

app = Flask(__name__)

@app.route("/")
def handle_root():
    return "Welcome to API"

@app.route("/health")
def handle_health():
    return "ok"

@app.route("/echo", methods=["POST"])
def handle_echo():
    return request.data, 200

@app.errorhandler(404)
def handle_404(e):
    return "not found", 404

app.run(port=8080)

File: tools/token_bench/programs/05_http_api/http_api.go
package main

import (
	"io"
	"net/http"
)

func main() {
	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
		if r.URL.Path != "/" {
			http.Error(w, "not found", http.StatusNotFound)
			return
		}
		w.Write([]byte("Welcome to API"))
	})

	http.HandleFunc("/health", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte("ok"))
	})

	http.HandleFunc("/echo", func(w http.ResponseWriter, r *http.Request) {
		body, _ := io.ReadAll(r.Body)
		w.Write(body)
	})

	http.ListenAndServe(":8080", nil)
}
